### PR TITLE
View solution page

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -9,9 +9,48 @@ import {
 const router = express.Router();
 
 router.get('/', async (req, res) => {
-
   res.render('home-page', {});
 });
+router.get('/view-solution/:solutionId', async (req, res) => {
+  const context = {
```
Comment (alibombaye) [inline app/routes.js]: I assume you are going to get this data from the API in a different PR?

File: app/views/sections/view-solution-description.njk [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      <div class="bc-c-title-block">
+        <h3 class="nhsuk-u-margin-bottom-2">Solution description</h3>
+      </div>
+      <div class="nhsuk-u-margin-bottom-4 nhsuk-u-padding-left-3">
+        {% if section.answers.summary %}
+          <div data-test-id="view-section-question-summary">
+              {{ viewQuestionTitle('summary', 'Summary') }}
+              {{ viewQuestionDataText('summary', section.answers.summary) }}
+          </div>
+        {% elseif section.answers.description %}
```
Comment (alibombaye) [inline app/views/sections/view-solution-description.njk]: I don't think this should be an elseif. It is possible to show both. It should depend on what data is provided from the API. If it is there then show it

File: app/views/view-solution-page.test.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  it('should render the foundation tag if isFoundation is true', (done) => {
+    const context = {
+      isFoundation: true,
+    };
+    const app = createDummyApp(context);
+    request(app)
+      .get('/')
+      .then((res) => {
+        const $ = cheerio.load(res.text);
+        const orgName = $('[data-test-id="solution-foundation-tag"]');
```
Comment (alibombaye) [inline app/views/view-solution-page.test.js]: change the name of the variable from orgName

File: app/views/view-solution-page.test.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  it('should not render the foundation tag if isFoundation is false', (done) => {
+    const context = {
+      isFoundation: false,
+    };
+    const app = createDummyApp(context);
+    request(app)
+      .get('/')
+      .then((res) => {
+        const $ = cheerio.load(res.text);
+        const orgName = $('[data-test-id="solution-foundation-tag"]');
```
Comment (alibombaye) [inline app/views/view-solution-page.test.js]: same here. Change name please

File: app/views/view-solution-page.test.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    const context = {
+      lastUpdated: 'some time',
+    };
+
+    const app = createDummyApp(context);
+    request(app)
+      .get('/')
+      .then((res) => {
+        const $ = cheerio.load(res.text);
+        const solutionName = $('[data-test-id="view-solution-page-last-updated"]');
```
Comment (alibombaye) [inline app/views/view-solution-page.test.js]: change name please

File: app/views/view-solution-page.test.js (hunk 3 above)
Comment (kat1906) [reply app/views/view-solution-page.test.js]: why?

Comment (kat1906) [reply app/views/view-solution-page.test.js]: oh right haha

File: app/views/sections/view-solution-description.njk (hunk 2 above)
Comment (kat1906) [reply app/views/sections/view-solution-description.njk]: ok it wasn't in the design and seemed a bit much to have the summary and description in there but can add back

Comment (alibombaye) [reply app/views/sections/view-solution-description.njk]: it should have both if both have been provided by the supplier. I think we are just a little ahead of the design at this point

Comment (kat1906) [reply app/views/sections/view-solution-description.njk]: whoop whoop 💃

File: app/routes.js (hunk 1 above)
Comment (kat1906) [reply app/routes.js]: was going to ... but I've linked it up now :)

File: app/controller.js [hunk 6]
```diff
@@ -1,31 +1,35 @@
...
+};
+
+export const getSolutionPageContext = async (solutionId) => {
+  const solutionData = await axios.get(`http://localhost:8080/api/v1/Solution/${solutionId}`);
+
+  const context = createSolutionPageContext(solutionData.data.solution);
+
+  return context;
+};
+
```
Comment (sam-roberts-bjss) [inline app/controller.js]: I removed the getSolutionPageContext. Are we still using this function?

File: app/styles/_app.scss [hunk 7]
```diff
@@ -22,4 +30,4 @@
     padding-right: 8px;
     padding-bottom: 4px;
     padding-left: 8px;
-}
\ No newline at end of file
+}
```
Comment (sam-roberts-bjss) [inline app/styles/_app.scss]: Once these changes are merged in, we will be able to remove this css class (govuk-tag).

Comment (kat1906) [reply app/styles/_app.scss]: yeah I'm working on extracting the foundation tag at the moment

File: app/controller.js (hunk 6 above)
Comment (kat1906) [reply app/controller.js]: probably not I will go through and delete the unused ones

File: integration-tests/showSolutionListPage.test.js [hunk 8]
```diff
@@ -6,6 +6,9 @@ const mocks = () => {
   nock('http://localhost:8080')
     .get('/api/v1/Solutions')
     .reply(200, aSolutionList);
+  nock('http://localhost:8080')
+    .get('/api/v1/Solutions/S1/Public')
+    .reply(200);
 };
```
Comment (sam-roberts-bjss) [inline integration-tests/showSolutionListPage.test.js]: I was thinking we would add this to the actual test as this mock is only specific to the click the view solution link test.

Comment (kat1906) [reply integration-tests/showSolutionListPage.test.js]: I only put it in to stop the error - you can change it if you like

File: integration-tests/showViewSolutionPage.test.js [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+test('should navigate to the link address when clicking on the link in the solution description', async (t) => {
+  pageSetup(t);
+  const getLocation = ClientFunction(() => document.location.href);
+  const solutionDescriptionLink = Selector('div[data-test-id="view-section-question-link"]');
+  await t
+    .click(solutionDescriptionLink)
+    .expect(getLocation())
+    .contains('link.com');
+});
```
Comment (sam-roberts-bjss) [inline integration-tests/showViewSolutionPage.test.js]: Does the test try to render link.com?

Comment (kat1906) [reply integration-tests/showViewSolutionPage.test.js]: what do you mean by render?

Comment (sam-roberts-bjss) [reply integration-tests/showViewSolutionPage.test.js]: Is there a website at link.com, do you see that page in the test browser

**Comment by sam-roberts-bjss** [comment]
All looks good :thumbsup:

File: app/styles/_app.scss (hunk 7 above)
Comment (sam-roberts-bjss) [reply app/styles/_app.scss]: Brilliant

**Pull request merged**